### PR TITLE
Use OrderedProperties to create logging annotation

### PR DESCRIPTION
**Pull request by chonton**
### Type of change
- Bugfix

### Description
Log properties should be ordered in the same fashion as other properties.  This removes last use of Properties class.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
Could you please review this @stanlyDoge? I think you did a lot of work on this area.

**Comment by chonton** [comment]
Can these changes be merged? Thanks!

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java [hunk 1]
```diff
@@ -111,7 +126,20 @@ public OrderedProperties addPair(String key, String value) {
...
+        return asPairs(null);
+    }
+
+    /**
+     * Generate a string with pairs formatted as key=value separated by newlines.  This string is
+     * expected to be used as an environment variable in a bash script and subsequently used in a bash
+     * <a href="https://www.tldp.org/LDP/abs/html/here-docs.html">here document</a> to create a
+     * properties file.
+     *
+     * @param comment A comment to add to the output
```
Comment (tombentley) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java]:
```suggestion
     * @param comment A comment to be prepended to the output, or null for no comment.
```

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java [hunk 2]
```diff
@@ -390,18 +428,25 @@ public String writeString() {
...
          *
          * @param bufferedWriter BufferedWriter to write values to.
          */
-        public void write(BufferedWriter bufferedWriter) throws IOException {
+        public void write(BufferedWriter bufferedWriter, String comment) throws IOException {
             this.bufferedWriter = bufferedWriter;
+            if (comment != null) {
+                bufferedWriter.write("# ");
+                bufferedWriter.write(comment);
+                bufferedWriter.newLine();
```
Comment (tombentley) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java]: So what if the comment contained `\n`?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java [hunk 3]
```diff
@@ -111,7 +126,20 @@ public OrderedProperties addPair(String key, String value) {
...
+    /**
+     * Generate a string with pairs formatted as key=value separated by newlines.  This string is
+     * expected to be used as an environment variable in a bash script and subsequently used in a bash
+     * <a href="https://www.tldp.org/LDP/abs/html/here-docs.html">here document</a> to create a
+     * properties file.
+     *
+     * @param comment A comment to add to the output
+     * @return String with one or more lines containing key=value pairs with the configuration options.
+     */
+    public String asPairs(String comment) {
```
Comment (tombentley) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java]: `asPairsWithComment()` would make the call sites more self-explanatory.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java (hunk 2 above)
Comment (chonton) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/OrderedProperties.java]: Now handling this case

**Comment by chonton** [comment]
Anything else needed?

**Comment by tombentley** [comment]
Thanks @chonton!

**Pull request merged**